### DISCUSSION TITLE
Is there any possibility of firejail getting released for Android?

### DISCUSSION THREAD

**Discussion by firejailaddssecuirty** [netblue30/firejail#4858] [Ideas]
Hi,
I purchased my first smartphone just a week ago. The model is Nokia G20. I use all network facing apps on my desktop inside a firejail sandbox. Is there any possibility of firejail getting released for Android too ? I have adopted the same policy with my phone that I follow on my desktop running EndeavourOS. Nothing gets installed outside of the repos. So it will be super nice to find firejail in Google Play Store. Last but not the least a GUI for firecfg.

Is this even possible ?

**Comment by rusty-snake** [comment]
Do you really want to downgrade the security model of android to the one of (desktop-)linux?

**Reply by firejailaddssecuirty** [reply]
Please elaborate. You mean Android which is based on Linux is more secure than (desktop) Linux ? This is my first smartphone. I know nothing about Android.

**Reply by topimiettinen** [reply]
[Android Security Features](https://source.android.com/security/features) are pretty impressive. On Linux desktop, apps and their data aren't isolated from other apps without Firejail (or SELinux) but on Android there are multiple protections. The kernel may be the same, but the system (traditional UNIX file system layout, user log in with shell, /home shared for all app data, sandboxing is add-on and opt-in etc.) wasn't designed with that kind of security in mind.

**Comment by topimiettinen** [comment]
Please correct me if I'm wrong, but I don't think it's possible to package privileged apps written in C in Android at all. Instead the apps use Java and they are heavily sandboxed with very limited file system access and no capabilities. Injecting Firejail would probably need some kind of custom rebuild of Android itself and then it would probably be easier to modify the facility responsible for sandboxing apps directly. Maybe the only benefit Firejail (or the mod) could offer would be custom seccomp filters for each app (if the app launcher/JVM wasn't already using seccomp).
